### DISCUSSION TITLE
Make `Screen` and testing fixtures available as part of the package

### DISCUSSION THREAD

**Discussion by hofaflo** [zauberzeug/nicegui#2200] [Ideas / Feature Requests]
Hi! I've been using your awesome package to create a few small web apps and am loving it so far! To simplify testing those apps, I usually copy `tests/conftest.py` and `tests/screen.py` into the project. As this can quickly go out of sync, I wanted to ask if you would be open to provide the `Screen` class and the fixtures contained in `conftest.py` as part of the package (e.g. under `nicegui.testing`)?

**Comment by falkoschindler** [comment]
Hi @hofaflo, I really like the idea!

We probably only need to
- move screen.py into a new directory nicegui/nicegui/testing (together with an \_\_init\_\_.py),
- define `Screen.SCREENSHOT_DIR` independent of the relatively imported `TEST_DIR` (`Path('screenshots')` could be just fine, i.e. a "screenshot" folder in the current working directory), and
- show how to use the `Screen` class in a new example.

But wait! What about conftest.py? Should we provide it as well, or only some useful functions like `reset_globals()`?

**Reply by falkoschindler** [reply]
@hofaflo Sorry, I just noticed that you already mentioned conftest.py. But I'm not sure if it is useful for every user project. There might be additional project-specific fixtures that need be combined with NiceGUI's default conftest.py. How is this possible?

**Reply by rodja** [reply]
I think it is possible to distribute a `conftest.py` with reasonable defaults with NiceGUI and still allow the user to have their own `conftest.py`. [These files get merged by pytest automatically](https://docs.pytest.org/en/6.2.x/fixture.html#:~:text=You%20can%20have%20multiple%20nested%20directories/packages%20containing%20your%20tests%2C%20and%20each%20directory%20can%20have%20its%20own%20conftest.py%20with%20its%20own%20fixtures%2C%20adding%20on%20to%20the%20ones%20provided%20by%20the%20conftest.py%20files%20in%20parent%20directories.).

**Reply by falkoschindler** [reply]
I created PR #2245 to address this feature request.

**Reply by hofaflo** [reply]
Thank you for implementing this so quickly! :heart:

**Reply by rodja** [reply]
It's already available in the just released 1.4.8 version 🚀
